### PR TITLE
Checks for testlogin API to ensure no AD server change is done

### PR DESCRIPTION
**Pull request by prachidamle**
Using /v1-autn/testlogin AD server settings can be changed, we need to
ensure with new settings same admin user can be searched

Also added serviceaccount pwd check while enabling access control.

https://github.com/rancher/rancher/issues/9641

### CODE REVIEW

File: providers/ldap/ldap_client.go [hunk 1]
```diff
@@ -641,6 +651,51 @@ func (l *LClient) TestLogin(testAuthConfig *model.TestAuthConfig) (int, error) {
...
+	} else if len(result.Entries) > 1 {
+		return status, fmt.Errorf("Multiple users found for the username with new server settings")
+	}
+
+	entry := result.Entries[0]
+	if !l.hasPermission(entry.Attributes, &testAuthConfig.AuthConfig.LdapConfig) {
+		return status, fmt.Errorf("User is probably disabled in the new server settings")
+	}
+
+	userIdentity, err := l.attributesToIdentity(entry.Attributes, entry.DN, l.ConstantsConfig.UserScope)
```
Comment (mrajashree) [inline providers/ldap/ldap_client.go:686]: @prachidamle entry.DN is what gets assigned as externalId to the client.Identity{} formed in attributesToIdentity. Can we check if entry.DN equals accessToken here itself ?

Comment (prachidamle) [reply providers/ldap/ldap_client.go:686]: Reason to call attributesToIdentity  is to make sure  changes to schema like UserLoginField/UserObjectClass also get tested
We allow admins to edit any of the schema fields during "edit settings", any changes there should also get validated
One way is to check if we are able to read all attributes as per the schema entered

**Comment by mrajashree** [comment]
LGTM

**Comment by cjellick** [comment]
LGTM

**Pull request merged**